receiver/splunksearchapireceiver/receiver.go

```diff
@@ -0,0 +1,317 @@
+	if err != nil {
+		return err
+	}
+
+	// set cancel function
+	cancelCtx, cancel := context.WithCancel(ctx)
+	ssapir.cancel = cancel
+
+	// create storage client
+	storageClient, err := adapter.GetStorageClient(ctx, host, ssapir.config.StorageID, ssapir.id)
+	if err != nil {
+		return fmt.Errorf("failed to get storage client: %w", err)
+	}
+	ssapir.storageClient = storageClient
+
+	err = ssapir.initCheckpoint(cancelCtx)
+	if err != nil {
+		return fmt.Errorf("failed to initialize checkpoint: %w", err)
+	}
+	go ssapir.runQueries(cancelCtx)
+	return nil
+}
+
+func (ssapir *splunksearchapireceiver) Shutdown(ctx context.Context) error {
+	ssapir.logger.Debug("shutting down logs receiver")
+	if ssapir.cancel != nil {
+		ssapir.cancel()
+	}
+
+	if ssapir.storageClient != nil {
+		if err := ssapir.checkpoint(ctx); err != nil {
+			ssapir.logger.Error("failed checkpoint", zap.Error(err))
+		}
+		return ssapir.storageClient.Close(ctx)
+	}
+	return nil
+}
+
+func (ssapir *splunksearchapireceiver) runQueries(ctx context.Context) {
+	for _, search := range ssapir.config.Searches {
+		// set current search query
+		ssapir.checkpointRecord.Search = search.Query
+
+		// set default event batch size (matches Splunk API default)
+		if search.EventBatchSize == 0 {
+			search.EventBatchSize = splunkDefaultEventBatchSize
+		}
+
+		// create search in Splunk
+		searchID, err := ssapir.createSplunkSearch(search)
+		if err != nil {
+			ssapir.logger.Error("error creating search", zap.Error(err))
+			return
+		}
+
+		// wait for search to complete
+		if err = ssapir.pollSearchCompletion(ctx, searchID); err != nil {
+			ssapir.logger.Error("error polling for search completion", zap.Error(err))
+			return
+		}
+
+		for {
+			ssapir.logger.Info("fetching search results")
+			results, err := ssapir.getSplunkSearchResults(searchID, offset, search.EventBatchSize)
+			if err != nil {
+				ssapir.logger.Error("error fetching search results", zap.Error(err))
+			}
+			ssapir.logger.Info("search results fetched", zap.Int("num_results", len(results.Results)))
+
+			// parse time strings to time.Time
+			earliestTime, err := time.Parse(time.RFC3339, search.EarliestTime)
+			if err != nil {
+				ssapir.logger.Error("earliest_time failed to be parsed as RFC3339", zap.Error(err))
+			}
+
+			latestTime, err := time.Parse(time.RFC3339, search.LatestTime)
+			if err != nil {
+				ssapir.logger.Error("latest_time failed to be parsed as RFC3339", zap.Error(err))
+			}
+
+			logs := plog.NewLogs()
+			for idx, splunkLog := range results.Results {
+				if (idx+exportedEvents) >= search.Limit && search.Limit != 0 {
+					limitReached = true
+					break
+				}
+				// convert log timestamp to ISO 8601 (UTC() makes RFC 3339 into ISO 8601)
+				logTimestamp, err := time.Parse(time.RFC3339, splunkLog.Time)
+				if err != nil {
+					ssapir.logger.Error("error parsing log timestamp", zap.Error(err))
+					break
+				}
+				if logTimestamp.UTC().Before(earliestTime) {
```

Comment:
I think this is supposed to be
```suggestion
				if logTimestamp.UTC().Before(earliestTime.UTC()) {
```
right?